mood-analysis.rb

```diff
@@ -19,6 +24,34 @@ def analyze_mood(words)
   return ":-|"
 end
 
+def happy_days(text)
+  smile_count = 0
+  day_count = 0
+  text.each do |line|
+    day_count += 1
+    if analyze_mood(line) == ":-)"
+      smile_count += 1
+    end
+    if smile_count == 3
+      break
+    end
+  end
+  if smile_count == 3
+    return "It took #{day_count} days for happiness to occur."
+  else
+    return "Sorry, no happy days..."
```

Comment:
If we did what I stated above, then here you don't need an if (else) at all because the only way we get here is that 3 happy days weren't found.